src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java

```diff
@@ -0,0 +1,113 @@
+package com.internet.shop.dao.jdbc;
+
+import com.internet.shop.dao.ProductDao;
+import com.internet.shop.exceptions.DataProcessingException;
+import com.internet.shop.lib.Dao;
+import com.internet.shop.model.Product;
+import com.internet.shop.util.ConnectionUtil;
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ProductDaoJdbcImpl implements ProductDao {
+    @Override
+    public Product create(Product item) {
+        String query = "INSERT INTO products (name, price) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement insert =
+                    connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS);
+            insert.setString(1, item.getName());
+            insert.setBigDecimal(2, item.getPrice());
+            insert.executeUpdate();
+            ResultSet generatedKey = insert.getGeneratedKeys();
+            while (generatedKey.next()) {
+                item.setId(generatedKey.getLong(1));
+            }
+            return item;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create product", e);
+        }
+    }
+
+    @Override
+    public Optional<Product> get(Long id) {
+        String query = "SELECT product_id, name, price FROM products "
+                + "WHERE product_id = ? AND deleted = 0";
```

Comment:
```suggestion
                + "WHERE product_id = ? AND deleted = FALSE";
```
this will make code easier to understand

Reply:
It can't be done, since TINYINT stands for BOOLEAN in MySQL.

Reply:
done